File: src/spec/resources/testprojects/nodetest/api/pet-api.ts
import type { ApiClient } from '../ApiClient.js';
import { BaseApi } from './BaseApi.js';
import { Configuration } from '../Configuration.js';
import { ObjectSerializer } from '../ObjectSerializer.js';
import type { Pet } from '../models/index.js';
import { PetFromJSON, PetToJSON } from '../models/index.js';

/**
 * PetApi provides methods for the Pet API group.
 */
export class PetApi extends BaseApi {
  constructor(config?: Configuration, apiClient?: ApiClient) {
    super(config, apiClient);
  }

  /**
   * Add a new pet to the store
   * @param pet Create a new pet in the store (required)
   * @return Pet
   */
  async addPet(pet: Pet): Promise<Pet> {
    if (pet == null) {
      throw new Error('Missing required parameter "pet" when calling addPet');
    }
    const path = `/pet`;
    const queryParams: Record<string, any> = {};
    const headerParams: Record<string, string> = {};
    return this.invokeApi(
      'POST',
      path,
      queryParams,
      headerParams,
      ObjectSerializer.serialize(pet, PetToJSON),
      ['application/json'],
      'application/json',
      (json: any) => ObjectSerializer.deserialize(json, PetFromJSON)
    ) as Pet;
  }

  /**
   * Deletes a pet
   * @param petId Pet id to delete (required)
   */
  async deletePet(petId: number): Promise<void> {
    if (petId == null) {
      throw new Error('Missing required parameter "petId" when calling deletePet');
    }
    const path = `/pet/{petId}`.replace(`{${'petId'}}`, encodeURIComponent(ObjectSerializer.toPathValue(petId)));
    const queryParams: Record<string, any> = {};
    const headerParams: Record<string, string> = {};
    (await this.invokeApi('DELETE', path, queryParams, headerParams, null, [], 'application/json', null)) as void;
  }

  /**
   * Finds Pets by status
   * @param status Status values that need to be considered for filter (optional)
   * @return Array<Pet>
   */
  async findPetsByStatus(status?: FindPetsByStatusStatusEnum): Promise<Array<Pet>> {
    const path = `/pet/findByStatus`;
    const queryParams: Record<string, any> = {};
    if (status != null) {
      queryParams['status'] = ObjectSerializer.toQueryValue(status);
    }
    const headerParams: Record<string, string> = {};
    return this.invokeApi(
      'GET',
      path,
      queryParams,
      headerParams,
      null,
      ['application/json'],
      'application/json',
      (json: any) => ObjectSerializer.deserializeArray(json, PetFromJSON)
    ) as Array<Pet>;
  }

  /**
   * Returns a single pet
   * Find pet by ID
   * @param petId ID of pet to return (required)
   * @return Pet
   */
  async getPetById(petId: number): Promise<Pet> {
    if (petId == null) {
      throw new Error('Missing required parameter "petId" when calling getPetById');
    }
    const path = `/pet/{petId}`.replace(`{${'petId'}}`, encodeURIComponent(ObjectSerializer.toPathValue(petId)));
    const queryParams: Record<string, any> = {};
    const headerParams: Record<string, string> = {};
    return this.invokeApi(
      'GET',
      path,
      queryParams,
      headerParams,
      null,
      ['application/json'],
      'application/json',
      (json: any) => ObjectSerializer.deserialize(json, PetFromJSON)
    ) as Pet;
  }

  /**
   * Update an existing pet
   * @param petId ID of pet to update (required)
   * @param pet Pet object that needs to be updated (required)
   * @return Pet
   */
  async updatePet(petId: number, pet: Pet): Promise<Pet> {
    if (petId == null) {
      throw new Error('Missing required parameter "petId" when calling updatePet');
    }
    if (pet == null) {
      throw new Error('Missing required parameter "pet" when calling updatePet');
    }
    const path = `/pet/{petId}`.replace(`{${'petId'}}`, encodeURIComponent(ObjectSerializer.toPathValue(petId)));
    const queryParams: Record<string, any> = {};
    const headerParams: Record<string, string> = {};
    return this.invokeApi(
      'PUT',
      path,
      queryParams,
      headerParams,
      ObjectSerializer.serialize(pet, PetToJSON),
      ['application/json'],
      'application/json',
      (json: any) => ObjectSerializer.deserialize(json, PetFromJSON)
    ) as Pet;
  }
}

/**
 * @export
 */
export const FindPetsByStatusStatusEnum = {
  Available: 'available',
  Pending: 'pending',
  Sold: 'sold',
  UnknownDefaultOpenApi: '11184809'
} as const;
export type FindPetsByStatusStatusEnum = (typeof FindPetsByStatusStatusEnum)[keyof typeof FindPetsByStatusStatusEnum];

File: src/spec/resources/testprojects/pytest/petstore_client/api/pet_api.py
from typing import Any, Dict, List, Optional, Union
from urllib.parse import quote

from petstore_client.models.pet import Pet

from ..api_client import ApiClient
from ..default_api_client import DefaultApiClient
from ..configuration import Configuration
from .base_api import BaseApi
from ..object_serializer import ObjectSerializer
from ..auth.authenticator import Authenticator


class PetApi(BaseApi):
    """PetApi provides methods for the pet API group."""

    def __init__(
        self,
        api_client: Optional[ApiClient] = None,
        config: Optional[Configuration] = None,
    ):
        super().__init__(api_client, config)

    def add_pet(
        self,
        auth: Authenticator,
        pet: Pet,
    ) -> Pet:
        """Add a new pet to the store
        :param auth: authenticator for this operation
        :param pet: Create a new pet in the store (required)
        :return: Pet
        """
        if pet is None:
            raise ValueError("Missing the required parameter 'pet' when calling add_pet")
        path = '/pet'
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = pet

        return self._invoke_api(
            'POST',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'Pet',
            auth,
        )

    def delete_pet(
        self,
        auth: Authenticator,
        pet_id: int,
    ) -> None:
        """Deletes a pet
        :param auth: authenticator for this operation
        :param pet_id: Pet id to delete (required)
        """
        if pet_id is None:
            raise ValueError("Missing the required parameter 'pet_id' when calling delete_pet")
        path = '/pet/{petId}'
        path = path.replace('{' + 'petId' + '}', quote(ObjectSerializer.to_path_value(pet_id), safe=''))
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = None

        return self._invoke_api(
            'DELETE',
            path,
            query_params,
            header_params,
            body,
            [],
            'application/json',
            None,
            auth,
        )

    def find_pets_by_status(
        self,
        status: Optional[str] = None,
    ) -> List[Pet]:
        """Finds Pets by status
        :param status: Status values that need to be considered for filter (optional, default to available)
        :return: List[Pet]
        """
        path = '/pet/findByStatus'
        query_params: Dict[str, Any] = {}
        if status is not None:
            query_params['status'] = ObjectSerializer.to_query_value(status)
        header_params: Dict[str, str] = {}
        body = None

        return self._invoke_api(
            'GET',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'List[Pet]',
            None,
        )

    def get_pet_by_id(
        self,
        pet_id: int,
    ) -> Pet:
        """Find pet by ID
        Returns a single pet
        :param pet_id: ID of pet to return (required)
        :return: Pet
        """
        if pet_id is None:
            raise ValueError("Missing the required parameter 'pet_id' when calling get_pet_by_id")
        path = '/pet/{petId}'
        path = path.replace('{' + 'petId' + '}', quote(ObjectSerializer.to_path_value(pet_id), safe=''))
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = None

        return self._invoke_api(
            'GET',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'Pet',
            None,
        )

    def update_pet(
        self,
        pet_id: int,
        pet: Pet,
    ) -> Pet:
        """Update an existing pet
        :param pet_id: ID of pet to update (required)
        :param pet: Pet object that needs to be updated (required)
        :return: Pet
        """
        if pet_id is None:
            raise ValueError("Missing the required parameter 'pet_id' when calling update_pet")
        if pet is None:
            raise ValueError("Missing the required parameter 'pet' when calling update_pet")
        path = '/pet/{petId}'
        path = path.replace('{' + 'petId' + '}', quote(ObjectSerializer.to_path_value(pet_id), safe=''))
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = pet

        return self._invoke_api(
            'PUT',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'Pet',
            None,
        )
